Mods/src/PuppiMod.cc

```diff
@@ -141,235 +160,222 @@ void PuppiMod::SlaveBegin()
-    IndicesF[i0] = i0;
-    IndicesC[i0] = i0;
+  for (UInt_t iCand = 0; iCand < numCandidates; iCand++) {
+    const PFCandidate *iCandidate = pfCandidates->At(iCand);
 
     Int_t etaBin = GetEtaBin(iCandidate);
-    if (etaBin < 0) 
+    if (etaBin < 0)
       continue;
-    if (iCandidate->Pt() < fMinPts[etaBin]) 
+    if (iCandidate->Pt() < fMinPts[etaBin])
       continue;                                               // if not meeting Pt cut, say it's PU
 
     // Determine if the PFCandidate is charged PU. This will help characterize neutral PU.
-    Int_t iParticleType = GetParticleType(iCandidate);
+    ParticleType iParticleType = GetParticleType(iCandidate, pv);
     // Mapper is getting nearby PFCandidates
-    std::vector<Int_t> nearList = Mapper->GetSurrounding(i0);
-    for (UInt_t i1 = 0; i1 < nearList.size(); i1++) {
-      Int_t i2 = nearList[i1];                                // Getting index from mapper results
-      if (i0 == i2) 
+    std::vector<UInt_t> nearList(fMapper->GetSurrounding(iCand));
+    for (UInt_t iNeighbor : nearList) {
+      if (iCand == iNeighbor)
         continue;                                             // Don't bother comparing a particle to itself
-      const PFCandidate *jCandidate = fPFCandidates->At(i2);
+
+      const PFCandidate *jCandidate = pfCandidates->At(iNeighbor);
       Double_t dRTemp = MathUtils::DeltaR(iCandidate,jCandidate);
       if (dRTemp > fRMin && dRTemp < fR0) {                                          // Only look at other particles in this range
-        Int_t jParticleType = GetParticleType(jCandidate);
+        ParticleType jParticleType = GetParticleType(jCandidate, pv);
         Double_t theAddition = (pow(jCandidate->Pt(),fAlpha))/(pow(dRTemp,fBeta));   // This is the thing we have to add inside the log
-        alphaF[i0] = alphaF[i0] + theAddition;                                       // First do the sum inside the log (alphaF)
-        if (jParticleType == 1)                                                      // if the particle is charged PV
-          alphaC[i0] = alphaC[i0] + theAddition;                                     //   add to alphaC
+        alphaF[iCand] += theAddition;                                       // First do the sum inside the log (alphaF)
+        if (jParticleType == kChargedPrimary)                                                      // if the particle is charged PV
+          alphaC[iCand] += theAddition;                                     //   add to alphaC
       }
     }
-    if (alphaF[i0] == 0) 
-      alphaF[i0] = -100;                                                             // Take the logs and ignore sum == 0 particles
-    else 
-      alphaF[i0] = TMath::Log(alphaF[i0]);
-    if (alphaC[i0] == 0) 
-      alphaC[i0] = -100;
-    else 
-      alphaC[i0] = TMath::Log(alphaC[i0]);
-    if (iParticleType == 2) {                                       // if charged PU, we might store it in the proper eta bin
+
```

Comment:
Add a spece after if-else, otherwise it looks like a single if-else if-else block at first sight.